Mlem/Views/Shared/Posts/Feed Post.swift

```diff
@@ -91,10 +91,21 @@ struct FeedPost: View {
     
     var barThickness: CGFloat { !postModel.read && diffWithoutColor && readMarkStyle == .bar ? CGFloat(readBarThickness) : .zero }
     var showCheck: Bool { postModel.read && diffWithoutColor && readMarkStyle == .check }
+    
+    var menuFunctions: [MenuFunction] {
+        let isMod = siteInformation.moderatedCommunities.contains(postModel.community.communityId)
+        
+        return postModel.menuFunctions(
+            editorTracker: editorTracker,
+            postTracker: postTracker,
+            community: isMod ? postModel.community : nil,
+            modToolTracker: isMod ? modToolTracker : nil
+        )
+    }
 
     var body: some View {
-        // this allows post deletion to not require tracker updates
-        if postModel.post.deleted {
+        // this allows post deletion/removal to not require tracker updates
+        if postModel.post.deleted || postModel.post.removed {
```

Comment:
Moderators are unable to restore posts because of this check. Maybe we should only hide removed posts for non-moderators, and show a "removed" label for moderators?

Reply:
Implemented 👍 

The API still doesn't return removed posts, though, so this is only accessible until the feed refreshes or via the modlog when that is implemented.